src/webp/decoder.rs

```diff
@@ -123,14 +123,27 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for WebpDecoder<R> {
     }
 
     fn colortype(&self) -> color::ColorType {
-        color::ColorType::Gray(8)
+        color::ColorType::RGB(8)
     }
 
     fn into_reader(self) -> ImageResult<Self::Reader> {
-        Ok(WebpReader(Cursor::new(self.frame.ybuf), PhantomData))
+        Ok(WebpReader(Cursor::new(self.read_image()?), PhantomData))
     }
 
     fn read_image(self) -> ImageResult<Vec<u8>> {
-        Ok(self.frame.ybuf)
+        let mut res = Vec::with_capacity(self.frame.ybuf.len() * 3);
+        for i in 0..self.frame.ybuf.len() {
+            let y = self.frame.ybuf[i] as f32;
+            let u = self.frame.ubuf[i] as f32;
+            let v = self.frame.vbuf[i] as f32;
+
+            let r = 298.082 * y / 256. + 408.583 * u / 256. - 222.291;
+            let g = 298.082 * y / 256. - 100.291* v / 256. - 208.120 * u / 256. + 135.576;
+            let b = 298.082 * y / 256. + 516.412 * v / 256. - 276.836;
```

Comment:
Where do these come from? It would be helpful to get a reference to something here. From googling the floating point numbers, I'd guess: https://www.microsemi.com/document-portal/doc_view/135317-ug0639-color-space-conversion-user-guide

Reply:
Seems to be from the ITU-R BT.601 standard

https://en.m.wikipedia.org/wiki/YCbCr

Reply:
Definitely not directly: https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.601-7-201103-I!!PDF-E.pdf

Reply:
Furthermore, 'Wikipedia' leaves out the meaining of 'corresponding RGB space' which if you carefully compare the xy-values of blue and green, is not the same as the RGB space defined for `sRGB`/`rec.709`.

Reply:
In any case, I would have expected the numbers from the section 'without any roundings' to appear here, as we don't have any problems with better precision when we use `f32` anyways.

Reply:
The numbers come from the linked Wikipedia page, however I will have a look at the standard and move this to a dedicated module with references to the standard.